packages/integration-sdk-runtime/src/execution/jobState.ts

```diff
@@ -158,6 +174,22 @@ export function createStepJobState({
     iterateRelationships: (filter, iteratee) =>
       graphObjectStore.iterateRelationships(filter, iteratee),
 
-    flush: () => graphObjectStore.flush(),
+    flush: () =>
+      graphObjectStore.flush(
+        async (entities) =>
+          uploader?.enqueue({
+            entities,
+            relationships: [],
+          }),
+        async (relationships) =>
+          uploader?.enqueue({
+            entities: [],
+            relationships,
+          }),
+      ),
+
+    async waitUntilUploadsComplete() {
+      await uploader?.waitUntilUploadsComplete();
```

Comment:
I may not have read the code correctly, but it does seem since `uploader.enqueue` is async and calls `await queue.onIdle()`, this will always return right away...

Reply:
I don't think that would be the case. The `enqueue` method is, in fact, `async`, but waiting for the enqueued function to actually settle is not async. The only reason why the `enqueue` function is async is so that we can throttle how many functions we are pushing into our queue. Throttling the `enqueue` will prevent us from buffering too many entities and relationships into memory while the uploads haven't actually completed. 

See: https://github.com/JupiterOne/sdk/pull/396/files#diff-38e59a3db7780e509c55171310fde11a5d109af248e2b859c4f0378ee65e049fR47